File: 백준/Silver/11723. 집합/집합.py
import sys
M = int(sys.stdin.readline())
S = set()
for i in range(M):
    arr = list(map(str, sys.stdin.readline().rstrip().split()))
    try:
        com, num = arr[0], int(arr[1])
    except:
        com = arr[0]

    if com == 'add':
        S.add(num)
    if com == 'remove' and num in S:
        S.remove(num)
    if com == 'check':
        if num in S:
            print(1)
        else:
            print(0)
    if com == 'toggle':
        if num in S:
            S.remove(num)
        else:
            S.add(num)
    if com == 'all':
        S = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20}
    if com == 'empty':
        S = set()

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

public class Main {
	static Set<Integer> S;
	 public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		int M = Integer.parseInt(br.readLine());
		S = new HashSet<>();
		StringTokenizer st  = null;
		for (int t = 0; t < M; t++) {
			st = new StringTokenizer(br.readLine());
			String order = st.nextToken();
			if (order.equals("add")) {
				S.add(Integer.parseInt(st.nextToken()));
			} else if (order.equals("remove")) {
				S.remove(Integer.parseInt(st.nextToken()));
			} else if (order.equals("check")) {
				if(S.contains(Integer.parseInt(st.nextToken()))) sb.append(1 + "\n");
				else sb.append(0 + "\n");
			} else if (order.equals("toggle")) {
				Integer num = Integer.parseInt(st.nextToken());
				if(S.contains(num)) S.remove(num);
				else S.add(num);
			} else if (order.equals("all")) {
				S = new HashSet<>(Arrays.asList(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20));
			} else if (order.equals("empty")) {
				S.clear();;
			} 
		}
		System.out.println(sb);
	}
}
